src/lib/booleans.js

```diff
@@ -0,0 +1,118 @@
+function negate(a) {
+  return !a;
+}
+
+function both(a, b) {
+  return a && b;
+}
+
+function either(a, b) {
+  return a || b;
+}
+
+function none(a, b) {
+  return !a && !b;
+}
+
+function one(a, b) {
+  if (!a && !b) {
+    return false;
+  }
+  return !a || !b;
+}
+function truthiness(a) {
+  if (a === 0) {
+    return false;
+  }
+  if (a === '') {
+    return false;
+  }
+  if (!a === !NaN) {
+    return false;
+  }
+  if (a == null) {
+    return false;
+  }
+  return true;
```

Comment:
The logic here is fine, but it can be condensed a lot using two NOT operators (!!) together to create a double negation. What this does is to first evaluate the value to a boolean that will be the opposite of what it actually is, eg passing in a string (which is a truthy value) will be made `false` by one of the ! operators, then the second ! will flip that to `true`, and so evaluating the string as being truthy. Nice trick that gets used more often than you might think!
```
  return !!a
```